src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -174,4 +160,14 @@ public async Task StartRequestAsync(TRequestContext? context, CancellationToken
         // so it can decide how to handle the result / exception.
         await _completionSource.Task.ConfigureAwait(false);
     }
+
+    public TTextDocumentIdentifier GetTextDocumentIdentifier<TTextDocumentIdentifier>(IMethodHandler handler)
+    {
+        if (handler is ITextDocumentIdentifierHandler<TRequest, TTextDocumentIdentifier> requestHandler)
+        {
+            return requestHandler.GetTextDocumentIdentifier(_request);
+        }
+
+        return default!;
```

Comment:
i don't understand this.  if this can happen, then the signature should be marked as nullable.

Reply:
Done